### DISCUSSION TITLE
Allow passing float values to setInterval method in BulkMessenger

### DISCUSSION THREAD

**Discussion by ghost** [nutgram/nutgram#304] [Q&A]
According to the explanation that [Telegram posted on the site](https://core.telegram.org/bots/faq#how-can-i-message-all-of-my-bot-39s-subscribers-at-once), we can send messages to almost 30 users every 1 second... and create a short break between them every time (for example, we can send messages to 25 users every 1 second with a break of 0.04 second message)

But the setInterval method in the BulkMessage class does not accept float type intervals! And the message will be sent much later!

![Screenshot from 2023-01-18 18-21-24](https://user-images.githubusercontent.com/78247242/213203190-bc224eae-107a-44a3-b2aa-a5ad952593bf.png)

--------------------------------------------------

![Screenshot from 2023-01-18 18-24-48](https://user-images.githubusercontent.com/78247242/213204229-e48c9878-750c-433a-a2cf-9c5e77356c22.png)

Please fix this issue.

**Comment by ghost** [comment]
No one pays attention to my posts ... : (
I wish I could develop Nutgram myself : /

**Reply by Lukasss93** [reply]
1. we work all day and don't always have time for everything 😅
2. did you know that you can also collaborate by opening a PR? 
3. when it was developed, the limits were different. But I still recommend sending a message every 2 seconds just in case.
4. we will work on it tomorrow 😉

**Reply by ghost** [reply]
Thank you. Yes, but currently my knowledge in this field is not enough to collaborate and develop Nutgram :/ 

Please see these posts and apply them in future updates... because they are very useful and needed :

https://github.com/nutgram/nutgram/discussions/297

https://github.com/nutgram/nutgram/discussions/299

Especially the ability to "send values to middlewares" :)

Thank you again bro
I visit Nutgram every day for new updates... (I have been following every day for a week now) 

:)

**Comment by Lukasss93** [comment]
We can't do this because the `pcntl_alarm` function only supports `int` seconds. 
We will also not change the call to the `sleep` function to keep the code consistent.
